js/map.js

```diff
@@ -96,42 +98,102 @@ const renderAdsOnMap = (ads) => {
     })
 };
 
-const getFilteredAds = (ads, typeHouses) => {
-  let filteredAds = ads;
-  if (typeHouses) {
-    filteredAds = ads.filter(ad => ad.offer.type.includes(typeHouses));
-  }
-  return filteredAds;
+
+const typeHousesFilterInput = document.querySelector('#housing-type');
+const pricesFilterInput = document.querySelector('#housing-price');
+const roomsFilterInput = document.querySelector('#housing-rooms');
+const guestsFilterInput = document.querySelector('#housing-guests');
+
+
+const priceFilterValues = {
+  LOW: 10000,
+  HIGH: 50000,
+};
+
+
+const validationTypeToFunction = {
+  checkType: function (ad) {
+    const valueType = typeHousesFilterInput.value;
+    return (ad.offer.type === valueType || valueType === 'any')
+  },
+
+  checkPrice: function (ad) {
+    const valuePrices = pricesFilterInput.value;
+    switch (valuePrices) {
+      case 'any':
+        return true;
+      case 'low': {
+        return ad.offer.price <= priceFilterValues.LOW
+      }
+      case 'middle': {
+        return ad.offer.price > priceFilterValues.LOW && ad.offer.price < priceFilterValues.HIGH;
+      }
+      case 'high': {
+        return ad.offer.price >= priceFilterValues.HIGH;
+      }
+      default:
+        return false;
+    }
+  },
+
+  checkRoomsNumber: function (ad) {
+    const valueRooms = roomsFilterInput.value;
+    return (ad.offer.rooms === Number(valueRooms) || valueRooms === 'any');
+  },
```

Comment:
Везде проверка на `any`. Как минимум нужно вынести any в костанту, а лучше дописать дополнительную функцию, что будет проверят:

```js
  const CONTROL_DEFAULT_VALUE = 'any';

  const checkIsControlInterrelation = function (controlValue, checker) {
    const isInterrelation = controlValue === CONTROL_DEFAULT_VALUE || checker;

    return isInterrelation;
  };
```

```suggestion
  checkRoomsNumber(ad) {
    return checkControlInterrelation(roomsFilterInput.value, offer.rooms === Number(roomsFilterInput.value))
  },
```